lib/scripts.js

```diff
@@ -444,6 +444,64 @@ var scripts = {
     ];
 
     return execScript.apply(scripts, args);
+  },
+
+  /**
+   * Attempts to retry a job
+   *
+   * @param {Job} job
+   *
+   * @return {Promise<Number>} Returns a promise that evaluates to a return code:
+   * 1 means the operation was a success
+   * 0 means the job does not exist
+   * -1 means the job is currently locked and can't be retried.
+   * -2 means the job was not found in the `failed` set
+   */
+  retryJob: function(job) {
+    var push = (job.opts.lifo ? 'R' : 'L') + 'PUSH';
+
+    var script = [
+      'if (redis.call("EXISTS", KEYS[1]) == 1) then',
+      '  if (redis.call("EXISTS", KEYS[2]) == 0) then',
+      '    if (redis.call("SREM", KEYS[3], ARGV[1]) == 1) then',
+      '      redis.call("' + push + '", KEYS[4], ARGV[1])',
+      '      redis.call("PUBLISH", KEYS[5], ARGV[1])',
+      '      return 1',
+      '    else',
+      '      return -2',
+      '    end',
+      '  else',
+      '    return -1',
+      '  end',
+      'else',
+      '  return 0',
+      'end'
+    ].join('\n');
```

Comment:
instead of this nested code I would propose something like this:
```lua
if ((redis.call("EXISTS", KEYS[1]) ~= 1) then
  return 0
end
if ((redis.call("EXISTS", KEYS[2])  ~= 0) then
  return -1
end
.
.
.
return 1;
```